server/src/main/java/org/elasticsearch/action/explain/ExplainRequest.java

```diff
@@ -59,24 +59,16 @@
     public ExplainRequest() {
     }
 
-    /**
-     * @deprecated Types are in the process of being removed. Use {@link ExplainRequest(String, String) instead.}
-     */
-    @Deprecated
-    public ExplainRequest(String index, String type, String id) {
-        this.index = index;
-        this.type = type;
-        this.id = id;
-    }
-
     public ExplainRequest(String index, String id) {
         this.index = index;
         this.id = id;
     }
 
     ExplainRequest(StreamInput in) throws IOException {
         super(in);
-        type = in.readString();
+        if (in.getVersion().before(Version.V_8_0_0)) {
```

Comment:
To be safe, we could read the type but assert that it is `_doc` (similarly to what we did in https://github.com/elastic/elasticsearch/pull/42198/).